versioned_docs/version-V3/guides/swaps/single-swaps.md

```diff
@@ -4,7 +4,14 @@ title: Single Swaps
 sidebar_position: 1
 ---
 
-## Setting up the Contract
+Swaps are the most common interaction with the Uniswap protocol. This example shows how to implement single swaps (for example, DAI/WETH9) using either an Exact Input or Exact Output swap. For simplification, the example hardcodes the token contract addresses, but as explained further below the contract could be modified to change pools and tokens on a per transaction basis.
```

Comment:
Thanks for the thoughtful suggestions. Regarding the exact input and exact output terms, I couldn't find an easy way to create a tooltip for those in the context of Docusaurus. There are some approached out there, but not super quick ones. (I'll create a doc enhancement issue for adding tooltip functionality.) So instead I cheated and looked at some other sites for ideas short of linking out to the Glossary, which is kind of excessive linking. The solution I used should be apparent in my commit. Devs should get the drift and pop open page search to go to the terms below (already described there) if they want to.

Regarding the introduction of the multi-hop examples in this intro, I don't think it's appropriate as this intro is for the single swap guide only. The multi-hop swap guide has its own intro. Best to keep them parallel and individual in that context.

Reply:
I'll add another issue as well for adding these terms to the Glossary.

Reply:
Sorry, when I said "...this is for the single swap guide only" I meant "article" not "guide". Incorporating intro info on multihop swaps would be appropriate for an overarching Overview article at the same TOC level as the Single Swaps and Multihop Swaps articles, but including that overarching info in just the Single Swaps article specifically isn't a good combination of outline structure and content. If that doesn't make sense, I'm happy to explain further and cite supporting resources and examples.

Reply:
@snreynolds  ...just a heads up that I'm working on a new commit that tries to incorporate your introduction of the function names themselves in the Single Swaps article intro copy. Using the function names is definitely starting to feel like a big improvement to me. I probably just had a knee jerk reaction to the introduction of the "multi-hop" swap stuff within the single-swap article page. Sorry about that.

For this PR, not sure how far we should go as seems we're already beyond the original goal for the PR, though for good reason. Lots of technical writing issues. For example, I'm seeing the potential to carve a separate Overview section instead (at an equal TOC level to the articles) that pulls the focus all the way out to let the developer know that... 

"The swap tutorials show you how to implement single-path and multi-path swap contracts using the external functions provided in the [ISwapRouter ](https://docs.uniswap.org/protocol/reference/periphery/interfaces/ISwapRouter) interface. 

You can implement a single-path swap that does such and such by using the external functions [exactInputSingle](https://docs.uniswap.org/protocol/reference/periphery/interfaces/ISwapRouter#exactinputsingle) and...

You can implement a multi-path swap does such and such by using ..."

**Notes on terminology suggestions:**

"path" or "hop" -- I used path because:

- "hop" is kind of a tcp/ip networking term
- the code example comments use "path"
- can't find other external uses of "hop" in this context, but there is one use of "path" in the V2 whitepaper (I searched all whitepaper versions for instances of "path" or "hop", but there was only one use of "path"...I should search more elsewhere...maybe later)

"exposed" or "external" -- either is fine really, kind of a nitpick, but I used "external" per https://docs.soliditylang.org/en/latest/contracts.html#visibility-and-getters, and just because it's generally good to be as precise as possible in dev doc. "exposed" could potentially be misinterpreted.

examples vs. guides vs. tutorials -- I think the "Guides" heading in the left nav TOC is kind of a misnomer as all of these are really "Tutorials" within a "Developer Guide", and that Developer Guide is everything (including "Concepts") above the "Technical Reference" which should be called "Reference Guide" instead of "Technical Reference".

Reply:
Actually, looking more closely, the Single Swap article doesn't really show how to use the external function exactInputSingle...isn't it instead showing how to create a function called swapExactInputSingle that uses the ExactInputSingleParams struct from ISwapRouter? I think that's more what the intro should say...will incorporate that into the Single Swaps commit I'm working on.

Reply:
oops...now I see the call to exactInputSingle down there as well! Sorry! Will re-adjust.

Reply:
Hi, seems like some of the suggestions haven't been committed yet. Going to commit those and then do a review.

Reply:
Sorry about that...I was actually just going through to doublecheck that, too, and I did notice I didn't update the title at the top of the local environment to Title Case...jeez...hope you can retain some of the improvements I made though. I'll stop my review and let you fix things, but am happy to help. Thanks, Sara.